distill/schemas/base.py

```diff
@@ -0,0 +1,40 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import abc
+
+from enum import Enum
+from typing import List, Optional
+
+from pydantic import AliasGenerator, BaseModel, Field
+from pydantic.alias_generators import to_camel
+from pydantic.config import ConfigDict
+
+
+class BaseSchema(BaseModel, abc.ABC):
+    """
+    Abstract base class to serve as model for any and all schemas
+    """
+
+    @property
+    @abc.abstractmethod
+    def _timestamp(self):
+        """
+        Represents a timestamp associated with the schema
+        Subclasses must override to provide specific timestamp value
+        Expected to return datetime object
```

Comment:
The expected return type is wrong.

It's expected to return: the result of `datetime.timestamp()` which per the docs is:
> Return POSIX timestamp corresponding to the [datetime](https://docs.python.org/3/library/datetime.html#datetime.datetime) instance. The return value is a [float](https://docs.python.org/3/library/functions.html#float) similar to that returned by [time.time()](https://docs.python.org/3/library/time.html#time.time).

We should add a `float` type annotation for the expected return type to make it unambiguous, and also explain that this is the POSIX time.